cmd/kubelet/app/server.go

```diff
@@ -644,6 +645,22 @@ func run(s *options.KubeletServer, kubeDeps *kubelet.Dependencies, stopCh <-chan
 			klog.Info("--cgroups-per-qos enabled, but --cgroup-root was not specified.  defaulting to /")
 			s.CgroupRoot = "/"
 		}
+
+		reservedSystemCPUs, err := cpuset.Parse(s.ReservedSystemCPUs)
+		if err != nil {
+			reservedSystemCPUs = cpuset.NewCPUSet()
+		}
+		if reservedSystemCPUs.Size() > 0 {
+			klog.Infof("KubeReserved=\"%v\", SystemReserved=\"%v\". The cpu will be overwritten by --reserved-cpus option", s.KubeReserved, s.SystemReserved)
+			if s.KubeReserved != nil {
+				delete(s.KubeReserved, "cpu")
+			}
+			if s.SystemReserved == nil {
+				s.SystemReserved = make(map[string]string)
+			}
+			s.SystemReserved["cpu"] = strconv.Itoa(reservedSystemCPUs.Size())
```

Comment:
So by overriding, this will ensure that you don’t reserve more than capacity with normal validation... the error message may be confusing as it will complain about system-reserved...

Reply:
Ack, will fix the error message.